### PR TITLE
Mo's branch 2

### PR DESCRIPTION
**Pull request by ombymo**
JWT token authentication added to habitRoutes, as well as userRoutes.

### CODE REVIEW

**Review by rozaromanova — CHANGES_REQUESTED** [review]
@ombymo I can't run your code, getting the following error:
```
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /Users/roza.romanova/Documents/TechLabs/habapp-backend/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/Users/roza.romanova/Documents/TechLabs/habapp-backend/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
```
I couldn't find why the reference is wrong right away, hence suggest you check your code and see why the package.json is either wrongly place, or addressed incorrectly. Both mine and Lasses' branches have it in habapp-backend/, yours is in habapp-backend/models

**Comment by ombymo** [comment]
Hey, it HAS TO be there (on same level as habapp-backend). I had it in the backend before, according to main, but it wouldn't run then. It's completely functional, all routes tested on postman, all routes secured with tokens as well.

**Comment by ombymo** [comment]
But CAN you merge? Because this branch was cloned via html, NOT SSH.

**Comment by ombymo** [comment]
@rozaromanova inside ../models there's only goalModel and userModel.

**Comment by rozaromanova** [comment]
@Gw3i can you maybe help out here? I can't run Mo's code, but apparently the other way around he can't run it. What's going wrong?

**Comment by rozaromanova** [comment]
@Codelikeneverbefore1 do you have a chance to run the app? I pushed the changes that I think should fix the problem for me, but it may break things for others...

**Comment by ombymo** [comment]
@rozaromanova IT IS FUNCTIONAL, ANYONE CAN RUN IT. Do u maybe need help to run it? Have you checked all necessary dependencies and packages are in place? did u start npm/async/etc. before running?

**Comment by ombymo** [comment]
Bildschirmfoto 2023-08-01 um 14 36 43
Bildschirmfoto 2023-08-01 um 14 36 54
Bildschirmfoto 2023-08-01 um 14 37 09
Bildschirmfoto 2023-08-01 um 14 37 54

**Comment by rozaromanova** [comment]
(tested how the deleting comments and reporting the behavior works, accidentally closed the branch)
comments for the reference:
https://d.pr/i/0D2MuQ
https://d.pr/i/GeRkZa
https://d.pr/i/VKwZ6v
https://d.pr/i/yxVBuH

File: habapp-backend/controllers/habitController.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        res.status(401)
+        throw new Error ('User not found')
+    }
+
+    if(goal.user.toString() !== user.id) { //trun goal id (user) to string, !== eq to user.id
+        res.status(401)
+        throw new Error('User not authorized')
+    }
+    
+    //console.log(habit)
```
Comment (Gw3i) [inline habapp-backend/controllers/habitController.js:82]: PLease remove

File: habapp-backend/controllers/habitController.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    const habit2 =     await Habit.deleteOne({ _id: req.params.id })
+    console.log(habit2)
+
+    //await Habit.deleteOne({ id: req.params.id })
```
Comment (Gw3i) [inline habapp-backend/controllers/habitController.js:86]: PLwase remove

File: habapp-backend/controllers/habitController.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    if(goal.user.toString() !== user.id) { //trun goal id (user) to string, !== eq to user.id
+        res.status(401)
+        throw new Error('User not authorized')
+    }
+    
+    //console.log(habit)
+    const habit2 =     await Habit.deleteOne({ _id: req.params.id })
+    console.log(habit2)
```
Comment (Gw3i) [inline habapp-backend/controllers/habitController.js:84]: Please remove

File: habapp-backend/controllers/userController.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            email: user.email,
+            token: generateToken(user._id)
+        })
+    } else {
+        res.status(400)
+        throw new Error('Invalid user data')
+    }
+
+
+    //res.json({message: 'Register User'})
```
Comment (Gw3i) [inline habapp-backend/controllers/userController.js:47]: Please remove

File: habapp-backend/controllers/userController.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    res.json({message: 'Login User'})
+})
+
+//desc: get user data
+//route GET /api/users/me
+//access: private
+const getMe = asyncHandler(async(req,res) => {
+    //res.json({message: 'User Data'})
```
Comment (Gw3i) [inline habapp-backend/controllers/userController.js:79]: Please remove

File: habapp-backend/models/userModel.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+const userSchema = mongoose.Schema({
+    name: {
+        type: String,
+        required: [true, 'Please add name']
+    },
+
+    email: {
+        type: String,
+        required: [true, 'Please add email']
+        //unique: true
```
Comment (Gw3i) [inline habapp-backend/models/userModel.js:11]: please remove

**Review by Gw3i — CHANGES_REQUESTED** [review]

File: habapp-backend/controllers/habitController.js (hunk 1 above)
Comment (ombymo) [reply habapp-backend/controllers/habitController.js:82]: und res.status(401) soll drin bleiben?

File: habapp-backend/controllers/userController.js (hunk 4 above)
Comment (ombymo) [reply habapp-backend/controllers/userController.js:47]: soll ich zeile 43 entfernen oder wie meinst du? den throw error teil

File: habapp-backend/controllers/habitController.js (hunk 1 above)
Comment (Gw3i) [reply habapp-backend/controllers/habitController.js:82]: Zeile 82

File: habapp-backend/controllers/habitController.js (hunk 2 above)
Comment (Gw3i) [reply habapp-backend/controllers/habitController.js:86]: Zeile 84 und 86

File: habapp-backend/controllers/userController.js (hunk 4 above)
Comment (Gw3i) [reply habapp-backend/controllers/userController.js:47]: Zeile 47

File: habapp-backend/controllers/userController.js (hunk 5 above)
Comment (Gw3i) [reply habapp-backend/controllers/userController.js:79]: Zeile 79

File: habapp-backend/models/userModel.js (hunk 6 above)
Comment (Gw3i) [reply habapp-backend/models/userModel.js:11]: Zeile 11

File: habapp-backend/server/index.js [hunk 7]
```diff
@@ -1,13 +1,27 @@
...
+
+app.use(errorHandler)
 
 app.listen(PORT, () => {
   console.log(`Server listening on ${PORT}`);
-});
\ No newline at end of file
+});
+
+//please leave as it is. it's all tested on postman already. if anything needs to be adapted for frontend let me know.
```
Comment (Gw3i) [inline habapp-backend/server/index.js:27]: Das auch weg, bitte

**Review by Gw3i — APPROVED** [review]

**Review by rozaromanova — APPROVED** [review]

**Pull request merged**